CompanyPaymentsProject/Company.cs

```diff
@@ -6,41 +6,31 @@
 
 namespace CompanyPaymentsProject
 {
-    class Company : Person
+    class Company
     {
-        
-        //lista osób i możliwość ich dodawania
-        List<Person> people = new List<Person>()
+        private List<Person> _persons = new List<Person>();
+       
+        public List<Person> GetPersonList()
         {
-            new Person { PersonID = 1, Name = "Jan", Surname = "Kowalski", Post.Assistant},
-            new Person { PersonID = 2, Name = "Anna", Surname = "Nowak", Post.Junior},
-            new Person { PersonID = 3, Name = "John", Surname = "Doe", Post.Manager},
-        };
-        public void AddNewWorker()
+            return _persons;
+        }
+
+        public void AddPayCheck(int personId, DateTime date)
         {
-            for (int i = people.Count; i < people.Count+1; i++)
-            {
-                people[i] = new Person();
-                Console.WriteLine("Enter worker #{0} name", i + 1 + people.Count);
-                people[i].Name = Console.ReadLine();
-                Console.WriteLine("Enter worker #{0} Surname", i + 1 + people.Count);
-                people[i].Surname =Console.ReadLine();
-                Console.WriteLine("Set the worker post in the company: 1. Manager, 2. Junior, 3. Assistant");
-                int workerPost = int.Parse(Console.ReadLine());
+            var person = _persons.SingleOrDefault(x => x.PersonID == personId);            
+            var salary = new SalaryCalculator().CalculateSalary(person.Post);
+            person.Paychecks.Add(new Paycheck { Salary = salary });
```

Comment:
Tutaj Ci opisywałem na Skype, tworzysz dwie płatności, a nie jedną, zerknij sobie i popraw i to chyba ostatnia rzecz :).